TestCertRotate cert access data race

```
=== RUN   TestSecurity/TestCertRotate
==================
WARNING: DATA RACE
Write at 0x00c000210750 by goroutine 109:
  kmesh.net/kmesh/pkg/controller/security.(*SecretManager).StoreCert()
      /kmesh/pkg/controller/security/manager.go:128 +0x247
  kmesh.net/kmesh/pkg/controller/security.(*SecretManager).fetchCert()
      /kmesh/pkg/controller/security/manager.go:210 +0x246
  kmesh.net/kmesh/pkg/controller/security.(*SecretManager).rotateCert.gowrap1()
      /kmesh/pkg/controller/security/manager.go:245 +0x4f

Previous read at 0x00c000210750 by goroutine 102:
  kmesh.net/kmesh/pkg/controller/security.runTestCertRotate()
      /kmesh/pkg/controller/security/manager_test.go:126 +0x6c5
  kmesh.net/kmesh/pkg/controller/security.TestSecurity.func2()
      /kmesh/pkg/controller/security/manager_test.go:43 +0x26
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1690 +0x226
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1743 +0x44

Goroutine 109 (running) created at:
  kmesh.net/kmesh/pkg/controller/security.(*SecretManager).rotateCert()
      /kmesh/pkg/controller/security/manager.go:245 +0x2f3
  kmesh.net/kmesh/pkg/controller/security.(*SecretManager).handleCertRequests()
      /kmesh/pkg/controller/security/manager.go:100 +0x314
  kmesh.net/kmesh/pkg/controller/security.(*SecretManager).Run.gowrap1()
      /kmesh/pkg/controller/security/manager.go:176 +0x44

Goroutine 102 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1743 +0x825
  kmesh.net/kmesh/pkg/controller/security.TestSecurity()
      /kmesh/pkg/controller/security/manager_test.go:42 +0x56
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1690 +0x226
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1743 +0x44
==================
    testing.go:1399: race detected during execution of test
=== RUN   TestSecurity/TestretryFetchCert
time="2025-01-08T07:28:16Z" level=error msg="fetchCert for [identity] error: abnormal test" subsys=security
--- FAIL: TestSecurity (4.61s)
    --- PASS: TestSecurity/TestBaseCert (0.30s)
    --- FAIL: TestSecurity/TestCertRotate (2.20s)
    --- PASS: TestSecurity/TestretryFetchCert (2.10s)
```

As we can see the race happened when test code does not access with a lock

--- a/pkg/controller/security/manager_test.go
+++ b/pkg/controller/security/manager_test.go
@@ -122,13 +122,14 @@ func runTestCertRotate(t *testing.T) {
 	// wait for rotate cert
 	time.Sleep(2 * time.Second)
 	for {
-		cert2 := secretManager.GetCert(identity1)
+		secretManager.certsCache.mu.RLock()
+		cert2 := secretManager.certsCache.certs[identity1]
 		if cert2 != nil && cert2.cert.CreatedTime != oldCert.CreatedTime {
-			secretManager.certsCache.mu.RLock()
 			newCert = *cert2.cert
 			secretManager.certsCache.mu.RUnlock()
 			break
 		}
+		secretManager.certsCache.mu.RUnlock()
 		time.Sleep(100 * time.Millisecond)
 	}
 

